private/syntax-path.rkt

```diff
@@ -564,6 +565,66 @@
       (check-regexp-match #rx"path is inconsistent" (exn-message thrown)))))
 
 
+(define (syntax-contains-path? init-stx path)
+  (with-handlers ([exn:fail? (λ (_) #false)])
+    (syntax-ref init-stx path)
+    #true))
```

Comment:
This masks other exception types. Instead of reusing `syntax-ref`, duplicate that code into this function and make this function return booleans directly instead of catching thrown exceptions.